src/codeu/chat/client/ClientMessage.java

```diff
@@ -165,11 +171,13 @@ private void showPreviousMessages(int count) {
    * 				Keyword to find in the messages.
    */
   public void findMessages(String keyword){
-	  
-	  for(final Message m : conversationContents){
-		  if(m.content.contains(keyword)){
+	  // Check the first element to determine if it the StoreLink is empty	  
+	  if(messagesSortedByKeyword.first(keyword) != null){
```

Comment:
Never mind, I get it now. You are checking to see if the keyword exists in the store.
Maybe a better name would be containsKeyword(...)?
So that you can check like this:
if(messageSortedByKeyword.containsKeyword(keyword)) {
   ...
}